src/main/java/nl/delftelectronics/spaceinvaders/core/entities/Barricade.java

```diff
@@ -9,6 +11,8 @@
 public class Barricade extends SpriteEntity implements Collidable {
     private static final String FILENAME = "/barricade.png";
 
+    private EventListenerList destroyedEvent;
```

Comment:
That would have been a possibility, but then I'd have to check the type of the Entity and cast it for the sake of code reuse.